src/modules/accommodations/components/filter-modal.tsx

```diff
@@ -0,0 +1,290 @@
+import Select from '@mui/material/Select';
+import TextField from '@mui/material/TextField';
+import Button from '@mui/material/Button';
+import Slider from '@mui/material/Slider';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogTitle from '@mui/material/DialogTitle';
+import Dialog from '@mui/material/Dialog';
+import Typography from '@mui/material/Typography';
+import IconButton from '@mui/material/IconButton';
+import Box from '@mui/material/Box';
+import Divider from '@mui/material/Divider';
+
+import { type AccommodationFilterParams } from '../types/accommodation-filter-params.type';
+import { accommdationSortByMap, AccommodationListSortBy } from '../constants/sort-by.constant';
+import { SortOrder, sortOrderMap } from '#/shared/constants/sort-order.constant';
+import { useListAccommodationQueryParams } from '../hooks/list-accommodations-query-params.hook';
+import { useSearchParams } from 'react-router-dom';
+import { amenitiesMap } from '../constants/amenities-map.constant';
+
+interface FilterModalProps {
+  open: boolean;
+  setOpen: (open: boolean) => void;
+}
+
+const styles = {
+  closeIcon: {
+    position: 'absolute',
+    left: '25px',
+    top: '25px',
+    cursor: 'pointer',
+  },
+
+  applyButton: {
+    color: '#fff',
+    fontWeight: '500',
+    fontSize: '16px',
+    backgroundColor: '#E91E63',
+    '&:hover': {
+      backgroundColor: '#CA1A55',
+    },
+  },
+
+  amenityButtons: (isSelected: boolean) => ({
+    border: '1px solid #C7C7C7',
+    borderColor: isSelected ? '#E91E63' : '#C7C7C7',
+    fontSize: '16px',
+    borderRadius: '20px',
+    backgroundColor: isSelected ? '#E91E63' : '#fff',
+    margin: '0 2px 2px 0',
+    color: isSelected ? '#fff' : '#C7C7C7',
+    '&:hover': {
+      borderColor: '#E91E63',
+      backgroundColor: '#E91E63',
+      color: '#fff',
+    },
+  }),
+
+  roomButton: (isSelected: boolean) => ({
+    flex: 1,
+    paddingY: '8px',
+    border: '2px solid #E91E63',
+    fontSize: '16px',
+    borderRadius: '20px',
+    backgroundColor: isSelected ? '#E91E63' : '#fff',
+    color: isSelected ? '#fff' : '#000',
+    '&:hover': {
+      backgroundColor: '#E91E63',
+      color: '#fff',
+    },
+  }),
+};
+
+export function AccommodationFilterModal({ open, setOpen }: FilterModalProps) {
+  const { setQueryParams } = useListAccommodationQueryParams();
+  const [_searchParams, setSearchParams] = useSearchParams();
+  const [filterParams, setFilterParams] = useState<AccommodationFilterParams>({
+    minPrice: 0,
+    maxPrice: 10000,
+    sortBy: AccommodationListSortBy.Price,
+    sortOrder: SortOrder.Asc,
+  });
+
+  const handlePriceRange = (newPrice: number | number[]) => {
+    const [minPrice, maxPrice] = newPrice as number[];
+    setFilterParams((prev) => ({
+      ...prev,
+      minPrice,
+      maxPrice,
+    }));
+  };
+
+  const handlePriceChange = (value: number, isMinPrice: boolean) => {
```

Comment:
How about instead of `isMinPrice: boolean`, we use `is: "min" | "max"`, which is less confusing and more readable?

Reply:
Fixed